core/src/main/java/org/geysermc/geyser/registry/loader/ResourcePackLoader.java

```diff
@@ -94,6 +102,33 @@ public Map<String, ResourcePack> load(Path directory) {
             resourcePacks.add(skullResourcePack);
         }
 
+        // Download CDN packs to get the pack uuid's
+        if (!Files.exists(CACHED_CDN_PACKS_DIRECTORY)) {
+            try {
+                Files.createDirectories(CACHED_CDN_PACKS_DIRECTORY);
+            } catch (IOException e) {
+                GeyserImpl.getInstance().getLogger().error("Could not create cached packs directory", e);
+            }
```

Comment:
We shouldn't continue with the new code below if this occurs - maybe move this stuff to a different method for an easy return & exit